src/app/add-item/add-item.component.ts

```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
 })
 export class AddItemComponent implements OnInit {
   dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+  createDialog: boolean;
```

Comment:
**SUGGESTION:** You could create enum like `ActionItemDialogMode` with values `Create` and `Update` and pass it as the input instead of this boolean. Then based on that you could have the single title value etc.

Reply:
I think that it would be better if you remove this variable and just pass the enum.